packages/wdio-browserstack-service/src/insights-handler.ts

```diff
@@ -323,14 +323,35 @@ class _InsightsHandler {
      */
 
     private attachHookData (context: any, hookId: string): void {
-        if (!context.currentTest || !context.currentTest.parent) {
+        if (context.currentTest && context.currentTest.parent) {
+            const parentTest = `${context.currentTest.parent.title} - ${context.currentTest.title}`
+            if (!this._hooks[parentTest]) {
+                this._hooks[parentTest] = []
+            }
+
+            this._hooks[parentTest].push(hookId)
             return
+        } else if (context.test) {
+            const setHooksFromSuite = (parent: any): boolean => {
```

Comment:
Let's add a null check & return false so that it doesn't break + recursive tree ends.